homeassistant/components/decora_wifi/common.py

```diff
@@ -0,0 +1,190 @@
+        self._password = password
+        self._iot_switches: dict[str, IotSwitch] = {}
+        self._logged_in = False
+
+    @property
+    def active_platforms(self) -> list[str]:
+        """Get the list of platforms which have devices defined."""
+        return [p for p in PLATFORMS if self._iot_switches[p]]
+
+    @property
+    def lights(self) -> list[IotSwitch]:
+        """Get the lights."""
+        return self._iot_switches[LIGHT_DOMAIN]
+
+    def _api_login(self):
+        """Log in to decora_wifi session."""
+        try:
+            user = self._session.login(self._email, self._password)
+            # If the call to the decora_wifi API's session.login returns None, there was a problem with the credentials.
+            if user is None:
+                raise LoginFailed
+            self._logged_in = True
+        except ValueError as exc:
+            raise CommFailed from exc
+        self._logged_in = True
+
+    def _api_logout(self):
+        """Log out of decora_wifi session."""
+        if self._logged_in:
+            try:
+                Person.logout(self._session)
+            except ValueError as exc:
+                raise CommFailed from exc
+        self._logged_in = False
+
+    def _api_get_devices(self):
+        """Update the device library from the API."""
+        perms: list[ResidentialPermission] = []
+        accounts: list[ResidentialAccount] = []
+        residences: list[Residence] = []
+        switches: list[IotSwitch] = []
+
+        # Gather permissions
+        try:
+            perms.extend(self._session.user.get_residential_permissions())
+        except ValueError as exc:
+            raise CommFailed from exc
+
+        # Gather residences for which the logged in user has permissions
+        for permission in perms:
+            if permission.residentialAccountId is not None:
+                accounts.append(
+                    ResidentialAccount(self._session, permission.residentialAccountId)
+                )
+            elif permission.residenceId is not None:
+                residences.append(Residence(self._session, permission.residenceId))
+        for acct in accounts:
+            try:
+                residences.extend(acct.get_residences())
+            except ValueError as exc:
+                raise CommFailed from exc
+
+        # Gather switches from residences
+        for res in residences:
+            try:
+                switches.extend(res.get_iot_switches())
+            except ValueError as exc:
+                raise CommFailed from exc
+
+        # Add the switches to the appropriate list in the iot_switches dictionary.
+        for switch in switches:
+            platform = DecoraWifiPlatform.classify_device(switch)
+            self._iot_switches[platform].append(switch)
+
+    def reauth(self):
+        """Reauthenticate this object's session."""
+        self._api_logout()
+        self._session = DecoraWiFiSession()
+        self._api_login()
+
```

Comment:
Does the integration use this code?

Reply:
Just checked, and it does not. YAGNI, eh?